pinax/stripe/actions/customers.py

```diff
@@ -45,29 +44,26 @@ def create(user, card=None, plan=settings.PINAX_STRIPE_DEFAULT_PLAN, charge_imme
         the pinax.stripe.models.Customer object that was created
     """
     trial_end = hooks.hookset.trial_period(user, plan)
-
     stripe_customer = stripe.Customer.create(
         email=user.email,
         source=card,
         plan=plan,
         quantity=quantity,
         trial_end=trial_end
     )
-    try:
-        with transaction.atomic():
-            cus = models.Customer.objects.create(
-                user=user,
-                stripe_id=stripe_customer["id"]
-            )
-    except IntegrityError:
-        # There is already a Customer object for this user
+    cus, created = models.Customer.objects.get_or_create(
+        user=user,
+        defaults={
+            "stripe_id": stripe_customer["id"]
+        }
+    )
+    if created:
+        sync_customer(cus, stripe_customer)
+        if plan and charge_immediately:
+            invoices.create_and_pay(cus)
+    else:
+        # remove this extra customer as it is not needed
         stripe.Customer.retrieve(stripe_customer["id"]).delete()
```

Comment:
Wouldn't it be better to create it only in case there is no local customer already in the first pace?

Reply:
Yes, that would make sense. Go for it!

Reply:
Cool, see https://github.com/pinax/pinax-stripe/pull/364.

Reply:
Just for reference: finally rejected in https://github.com/pinax/pinax-stripe/pull/364.